homeassistant/components/habitica/sensor.py

```diff
@@ -153,53 +239,59 @@ async def update(self):
 class HabitipySensor(SensorEntity):
     """A generic Habitica sensor."""
 
-    def __init__(self, name, sensor_name, updater):
+    _attr_has_entity_name = True
+    entity_description: HabitipySensorEntityDescription
+
+    def __init__(
+        self,
+        coordinator,
+        entity_description: HabitipySensorEntityDescription,
+        entry: ConfigEntry,
+    ) -> None:
         """Initialize a generic Habitica sensor."""
-        self._name = name
-        self._sensor_name = sensor_name
-        self._sensor_type = SENSORS_TYPES[sensor_name]
-        self._state = None
-        self._updater = updater
+        super().__init__()
+        if TYPE_CHECKING:
+            assert entry.unique_id
+        self.coordinator = coordinator
+        self.entity_description = entity_description
+        self._attr_unique_id = f"{entry.unique_id}_{entity_description.key}"
+        self._attr_device_info = DeviceInfo(
+            entry_type=DeviceEntryType.SERVICE,
+            manufacturer=MANUFACTURER,
+            model=NAME,
+            name=entry.data[CONF_NAME],
```

Comment:
What's the config entry title? If that is the same as Name, we could omit this line because a primary device without name uses the config entry title as name

Reply:
the integration does some weird things there, when setting up the integration, the title is "Default username", CONF_NAME is optional but later in the init if it was left empty, it gets updated with the profile name. Will definitively work on improving the config flow, but as you said, this PR was already starting to do too much. I think we should leave it like that for now, but I will keep it in mind for the next PRs

Reply:
Check!